plugins/recorder/src/screen-recorder.ts

```diff
@@ -0,0 +1,47 @@
+import { Recorder } from './recorder'
+import { TusUploader, Uploader, Options } from './uploader'
+
+export class ScreenRecorder {
+  private readonly recorder: Recorder
+  private readonly uploader: Uploader
+
+  constructor (recorder: Recorder, uploader: Uploader) {
+    this.recorder = recorder
+    this.uploader = uploader
+  }
+
+  static async fromNavigatorMediaDevices (opts: Options): Promise<ScreenRecorder> {
+    let width = 0
+    let height = 0
+    const combinedStream = new MediaStream()
+    const displayStream = await navigator.mediaDevices.getDisplayMedia({ video: { frameRate: opts.fps ?? 30 }, audio: true })
```

Comment:
you need to add some validation here that navigator exists and user granted access to devices, otherwise the code will result in error

Reply:
it's managing in UI components